src/app/mapView/MapView.jsx

```diff
@@ -32,41 +25,109 @@ export default class MapView extends Component {
     navigator.geolocation.getCurrentPosition(
       userPosition => {
         const { coords } = userPosition;
-        this.onCenterChanged({
-          lat: coords.latitude,
-          lng: coords.longitude,
+        this.setState({
+          center : {
+            lat: coords.latitude,
+            lng: coords.longitude,
+          }
         });
       },
       e => console.error('Failed to get current position', e),
       { timeout: geolocationTimeout },
     );
   }
+
   onSearchChanged = event => {
-    this.setState({ searchString: event.target.value }, () => {
-      this.fetchLocations();
-    });
+    const searchString = event.target.value;
+    if(searchString){ 
+      this.onSuggestionsFetchRequested({searchString});
+    } else {
+      this.onSuggestionsClearRequested();
+    }
+  };
+
+  onBoundsChanged = ({center, radius}) => {
+    if(center.lat() === this.state.center.lat && 
+        center.lng() === this.state.center.lng) return;
+    this.fetchLocations(center, radius);
+  }
+
+  onSuggestionsFetchRequested = ({ searchString, reason }) => {
+    getOrganizations(searchString)
+      .then(organizations => { 
+          this.setState({ suggestions : organizations })
+      })
+      .catch(e => console.error('error', e));
   };
 
-  onCenterChanged = center => {
-    this.setState({ center }, () => {
-      this.fetchLocations();
+  handleSuggestionClick = (organization) => {
+    getOrganizationLocations(organization.id)  
+      .then(locations => { 
+        if(!locations.length) return;
+        locations = locations.map( loc => ({...loc, Organization : organization}))   //add orgranization to locations
```

Comment:
Mutating arguments always feels a bit iffy to me, even if it is just re-assigning the reference and not risking any actual side-effects (I actually thought there was an eslint rule for that). Then again, adding a "locationsWithObject" isn't very pretty either, so I dunno.

Reply:
I can create a different variable name.